crates/utils/src/settings.rs

```diff
@@ -91,21 +164,14 @@ impl Settings {
   ///
   /// Note: The env var `LEMMY_DATABASE_URL` is parsed in
   /// `lemmy_db_queries/src/lib.rs::get_database_url_from_env()`
-  fn init() -> Result<Self, ConfigError> {
-    let mut s = Config::new();
-
-    s.merge(File::with_name(&Self::get_config_defaults_location()))?;
+  fn init() -> Result<Self, LemmyError> {
+    // Read the config file
+    let mut custom_config = from_str::<Settings>(&Self::read_config_file()?)?;
 
-    s.merge(File::with_name(&Self::get_config_location()).required(false))?;
+    // Merge with default
+    custom_config.merge(Settings::default());
 
-    // Add in settings from the environment (with a prefix of LEMMY)
-    // Eg.. `LEMMY_DEBUG=1 ./target/app` would set the `debug` key
-    // Note: we need to use double underscore here, because otherwise variables containing
-    //       underscore cant be set from environmnet.
-    // https://github.com/mehcode/config-rs/issues/73
-    s.merge(Environment::with_prefix("LEMMY").separator("__"))?;
```

Comment:
Apparently there's a way to do this but I haven't tried it yet: https://git.sr.ht/~ireas/merge-rs/tree/master/item/examples/args.rs#L32

Reply:
I think instead I'm just gonna replace our only use of those, which is in `federation/docker-compose.yml`, with `lemmy.hjson` mounts.

Reply:
I think configuration via env vars is extremely useful, and I definitely want to keep it.

Reply:
I don't know of a way to merge hierarchical env vars, but if we want to support specific env vars, that's trivial:

```rust
custom_config = {bleh_thing: env::var("bleh_thing"), ...custom_config};
```

Reply:
In the worst case we have to specify the env var name for each var manually :(